extensions/amp-bind/0.1/bind-expression.js

```diff
@@ -35,6 +36,58 @@ const BUILT_IN_FUNCTIONS = 'built-in-functions';
  * @const @private {!Object<string, !Object<string, Function>>}
  */
 const FUNCTION_WHITELIST = (function() {
+
+  /**
+   * Custom wrappers for mutating array methods so that we can offer the
+   * same functionality without allowing mutation on variables in bind scope.
+   */
+  const BindArrays = {
+    /**
+     * Similar to Array.prototype.splice, except it returns a copy of the
+     * passed-in array with the desired modifications.
+     * @param {!Array} array
+     * @param {number=} start
+     * @param {number=} deleteCount
+     * @param {...?} items
+     */
+    /*eslint "no-unused-vars": 0*/
+    'copyAndSplice': function(array, start, deleteCount, items) {
+      if (!isArray(array)) {
+        user().warn(
+            TAG,
+            `copyAndSplice: ${array} is not an array; returning null.`);
+        return null;
+      }
+      const copy = array.slice();
+      copy.splice.apply(copy, Array.prototype.slice.call(arguments, 1));
```

Comment:
You can use [Function.prototype.call](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/call) and use the params directly instead, which will save you a `slice` call.